programmer/tinyprog/__init__.py

```diff
@@ -419,16 +457,14 @@ def program_sectors(self, addr, data):
                         self.progress("FAILED!")
                         return False
 
-
         self.progress("Success!")
         return True
 
-
     def boot(self):
         try:
             self.ser.write(b"\x00")
             self.ser.flush()
-        except:
+        except Exception:
```

Comment:
@mithro is `writeTimeoutError` an exception?  I'm both unclear whether that's true, and also on one run I did have programming bail out due to a `writeTimeoutError` which I hadn't seen happen before.

Ewen

```

(tinyfpga) ewen@parthenon:/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer$ PYTHONPATH=. python3 /src/fpga/litex-buildenv/build/conda/bin/tinyprog --program-image /src/fpga/litex-buildenv/build/tinyfpga_bx_base_lm32.minimal//image-gateware+bios+micropython.bin

    TinyProg CLI
    ------------
    Using device id 1d50:6130
    Only one board with active bootloader, using it.

    Programming /dev/ttyACM0 with /src/fpga/litex-buildenv/build/tinyfpga_bx_base_lm32.minimal//image-gateware+bios+micropython.bin
    Programming at addr 028000
    Waking up SPI flash
    442884 bytes to program
    Programming and Verifying:   1%|        | 5.63k/443k [00:01<01:31, 4.80kB/s]
Traceback (most recent call last):
  File "/src/fpga/litex-buildenv/build/conda/bin/tinyprog", line 11, in <module>
    sys.exit(main())
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__main__.py", line 455, in main
    if not fpga.program_bitstream(addr, bitstream):
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__init__.py", line 497, in program_bitstream
    return self.program_sectors(addr, bitstream)
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__init__.py", line 441, in program_sectors
    max_length=256)
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__init__.py", line 403, in write
    self._write(addr, write_data)
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__init__.py", line 389, in _write
    self.wait_while_busy()
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__init__.py", line 319, in wait_while_busy
    while to_int(self.read_sts()) & 1:
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__init__.py", line 276, in read_sts
    return self.cmd(0x05, read_len=1)
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__init__.py", line 262, in cmd
    self.ser.write(bytearray(cmd_write_string))
  File "/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer/tinyprog/__init__.py", line 94, in write
    self.ser.write(data)
  File "/src/tinyfpga/tools/lib/python3.5/site-packages/serial/serialposix.py", line 579, in write
    raise writeTimeoutError
serial.serialutil.SerialTimeoutException: Write timeout
(tinyfpga) ewen@parthenon:/src/tinyfpga/packages/TinyFPGA-Bootloader/programmer$ 
```

Reply:
I can find:

https://chromium.googlesource.com/experimental/chromium/src/+/84aa73c4dd94ac0089a164a7f8e8b1571c7a5eb9/tools/telemetry/third_party/pyserial/serial/serialutil.py#108

which suggests `writeTimeoutError` is a `SerialTimeoutException`:

https://chromium.googlesource.com/experimental/chromium/src/+/84aa73c4dd94ac0089a164a7f8e8b1571c7a5eb9/tools/telemetry/third_party/pyserial/serial/serialutil.py#104

which seems to be a `SerialError`:

https://chromium.googlesource.com/experimental/chromium/src/+/84aa73c4dd94ac0089a164a7f8e8b1571c7a5eb9/tools/telemetry/third_party/pyserial/serial/serialutil.py#100

which seems to be an `IOError`.  Which I *think* should be an `Exception`?  So I'm a bit unclear what's going on here.

But if it's only `writeTimeoutError` that's expected, maybe it'd make sense to be more specific?  `except Exception` doesn't seem much more useful than `except: pass`....

Ewen

Reply:
We concluded this was a different path triggering this exception.

Reply:
(We should probably also be only catching writeTimeoutError here too as you mention.)